### PR TITLE
feat(physics): enhance the scroll physics behaviour

### PR DESCRIPTION
**Pull request by robertohuertasm**
With these changes the scroll physics will allow the page to recover to its center position even if
some of the movements are blocked. I've also added more tests and a nicer readme.

### CODE REVIEW

File: example/pubspec.lock [hunk 1]
```diff
@@ -109,13 +109,6 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "1.6.4"
-  pedantic:
```
Comment (daviddgl) [inline example/pubspec.lock:112]: Why did you remove `pedantic` package?

Comment (robertohuertasm) [reply example/pubspec.lock:112]: I didn't... see the pubspec.yaml... This file is automatically generated. Let me do `flutter pub get` again.

Comment (robertohuertasm) [reply example/pubspec.lock:112]: See, it is below. So it wasn't removed after all. Just reordered.

Comment (daviddgl) [reply example/pubspec.lock:112]: If I open the file I can't find any pedantic word in it... Am I missing something?

Comment (robertohuertasm) [reply example/pubspec.lock:112]: https://github.com/Alpha-health/horizontal_blocked_scroll_physics/blob/master/pubspec.lock#L98

Comment (robertohuertasm) [reply example/pubspec.lock:112]: Oh, this is the example project's pubspec.lock! I'm not sure why pedantic got removed. My bad! Sorry for the confusion! Anyway, I don't think it matters that much since this file is completely managed by Flutter and it's only about the example. It has no impact in the library at all.

Comment (robertohuertasm) [reply example/pubspec.lock:112]: I mean, I didn't even changed the `example/pubspec.yaml` at all so it probably was a dependency from another package.

Comment (daviddgl) [reply example/pubspec.lock:112]: 👍
